### PR TITLE
76 type docs scope update

### PR DESCRIPTION
**Pull request by nphias**
- updates to all package files (dev-dependencies)
- updates typedoc yml workflow (github actions)
- added documentation comments
- improved typedoc scope
- added UML feature typedoc
- added package exclusions to root tsconfig

### CODE REVIEW

File: packages/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+External Id Attestation
+-----------------------
+
+This feature allows a user to register an ID from an external system and given external verification the autority agent will make an attestation and associate the ID with a Holochain ``AgentPubKey``. Two additional fields (``display_name`` and ``request_id``) give the attestion object more uniqueness and traceability. A noticeble behaviour is that a user can register multiple external_id's against different AgentPubKeys. \
```
Comment (8e8b2c) [inline packages/README.md:4]: "autority" should be "authority"

File: packages/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+For more details on a user agent workflows see: [User Requestor Client](/classes/client_src.ExternalIdAttestationRequestorClient.html) \
+For the authority agent workflows see: [Authority Attestor Client](/classes/authority_src.ExternalIdAttestorClient.html) 
+
+
+Username Attestation
+--------------------
+
+This feature allows a user to register an username against a Holochain AgentPubKey with the attestation of the Authoriy agent. Usermetadata atrribute-links allow the user to attach any related metadata. \
```
Comment (8e8b2c) [inline packages/README.md:12]: "Authoriy"...

File: packages/authority/src/external-id-attestor/external-id-attestor-client.ts [hunk 3]
```diff
@@ -8,6 +8,20 @@ type ExternalIdAttestationRequested = PickByType<
...
 >;
 
+/**
+ * ```mermaid
+ * sequenceDiagram
+ *  participant H as Holochain
+ *  actor A as Authority
+ *  participant T as TokenAssessor
+ *  H->>A: Signal (ExternalIdAttestationRequested)
+ *  A->>T: exchangeAccessToken (Username, Password)
```
Comment (8e8b2c) [inline packages/authority/src/external-id-attestor/external-id-attestor-client.ts:18]: The authority swaps a `codeVerifier` (i.e. the user evidence that they started the web2 authorisation flow) and a `code` (i.e. the authorisation code issued by the web2 identity provider, which acts as secret and reference for the ongoing web2 authorisation flow) for the access token. The user's password never goes near the authority.

We've basically repurposed elements from this flow:
https://auth0.com/docs/get-started/authentication-and-authorization-flow/authorization-code-flow-with-pkce

File: packages/authority/src/index.ts [hunk 4]
```diff
@@ -1,4 +1,7 @@
 export * from "./external-id-attestor";
+export {ExternalIdAttestorClient} from "./external-id-attestor/external-id-attestor-client";
```
Comment (8e8b2c) [inline packages/authority/src/index.ts:2]: prettier please!

**Review by 8e8b2c — CHANGES_REQUESTED** [review]

File: packages/README.md (hunk 1 above)
Comment (nphias) [reply packages/README.md]: fixed, sorry have added a spell checker extension to my IDE

**Pull request merged**